obs/project.go

```diff
@@ -91,3 +96,110 @@ type RepositoryPath struct {
 	Project    string `json:"project" xml:"project,attr"`
 	Repository string `json:"repository" xml:"repository,attr"`
 }
+
+type Status struct {
+	XMLName xml.Name `json:"status" xml:"status"`
+	Code    string   `json:"code" xml:"code,attr"`
+	Summary string   `json:"summary" xml:"summary"`
+}
+
+type APIError struct {
+	HTTPStatusCode int
+	XMLStatusCode  string
+	Message        string
+}
+
+func (e *APIError) Error() string {
+	return fmt.Sprintf("HTTP status %d: %s (%s)", e.HTTPStatusCode, e.XMLStatusCode, e.Message)
+}
+
+func (c Client) CreateProject(project *Project) error {
+	xmlData, err := xml.MarshalIndent(project, "", " ")
+	if err != nil {
+		return err
+	}
+
+	urlPath := path.Join(c.APIURL, "source", project.Name, "_meta")
+	req, err := http.NewRequest("PUT", urlPath, bytes.NewBuffer(xmlData))
+	if err != nil {
+		return &APIError{
+			HTTPStatusCode: http.StatusInternalServerError,
+			XMLStatusCode:  "",
+			Message:        fmt.Sprintf("failed to generate a new request: %v", err),
+		}
+	}
+
+	req.SetBasicAuth(c.Username, c.Password)
+	req.Header.Set("Accept", "application/xml; charset=utf-8")
+
+	resp, err := c.Client.Do(req)
+	if err != nil {
+		return &APIError{
+			HTTPStatusCode: resp.StatusCode,
```

Comment:
There's no StatusCode for this type of errors. Errors covered by `err != nil` are usually connection errors (e.g. internet connectivity or inability to reach the API). The error here should be handled in the same way as described in the previous comment.